watertap/tools/analysis_tools/loop_tool/loop_tool.py

```diff
@@ -402,9 +402,18 @@ def get_diff_params(self, key, loop_value):
             force_rerun,
         )
 
+    def get_num_samples(self, value_dict):
+        num_samples = value_dict.get("num_samples")
+        if value_dict.get("num_samples") is not None:
+            return value_dict.get("num_samples")
+        elif value_dict.get("array") is not None:
+            return len(value_dict.get("array"))
```

Comment:
This isn't tested -- which surprises me given the test below.

Reply:
I don't think I have a test configured to run tests on predifined fixed or random sample in loop tool.  I guess I should add a test fo rthis.